nextstrain/cli/command/login.py

```diff
@@ -64,54 +80,73 @@ def register_parser(subparser):
 
             Password for nextstrain.org user.  Required if :option:`--no-prompt` is
             used without existing valid/renewable tokens.
+
+        If you want to suppress ever opening a web browser automatically, you
+        may set the environment variable ``NOBROWSER=1``.
         """)
 
     return parser
 
 
 def run(opts):
+    remote, url = parse_remote_path(opts.remote)
+    assert url.origin
+
     if opts.renew:
-        user = renew()
+        user = remote.renew(url.origin)
         if not user:
             raise UserError("Renewal failed or not possible.  Please login again.")
         return
 
-    user = current_user()
+    user = remote.current_user(url.origin)
 
     if not user:
         username = opts.username
         password = environ.get("NEXTSTRAIN_PASSWORD")
 
         if opts.no_prompt and (username is None or password is None):
-            raise UserError("No Nextstrain.org credentials found and --no-prompt prevents interactive login.")
+            raise UserError(f"No {url.origin} credentials found and --no-prompt prevents interactive login.")
 
-        print("Logging into Nextstrain.org…")
+        print(f"Logging into {url.origin}…")
         print()
 
-        if username is not None:
-            print(f"Username: {username}")
-        else:
-            username = prompt(getuser)
-
-        if password is not None:
-            print("Password: (from environment)")
+        # If we have either a username or a password, then obtain a complete
+        # set of credentials and do password-based login.
+        #
+        # If we have neither, we'll do browser-based login.
+        if username is not None or password is not None:
+            credentials = lambda: prompt_for_credentials(username, password)
         else:
-            password = prompt(getpass)
+            credentials = None
 
-        print()
-
-        user = login(username, password)
+        user = remote.login(url.origin, credentials)
         print()
     else:
         if opts.username is not None and opts.username != user.username:
             raise UserError(f"""
-                Login requested for {opts.username}, but {user.username} is already logged in.
+                Login requested for {opts.username}, but {user.username} is already logged in to {url.origin}.
                 
                 Please logout first if you want to switch users.
                 """)
 
-    print(f"Logged into nextstrain.org as {user.username}.")
-    print("Log out with `nextstrain logout`.")
+    print(f"Logged into {url.origin} as {user.username}.")
+    print(f"Log out with `nextstrain logout {shquote(url.origin)}`.")
```

Comment:
I would favor not printing the remote when it's the default (nextstrain.org).

Reply:
Hmm, I don't agree. I think because multiple remotes are possible, it's good to always be explicit to avoid confusing output.